tests/test_init_and_backend.py

```diff
@@ -0,0 +1,139 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import filecmp
+import tempfile
+from pathlib import Path
+
+import mkosi
+import mkosi.backend as backend
+import pytest
+
+
+def test_workspace():
+    assert backend.workspace(Path("/home/folder/mkosi/mkosi")) == Path("/home/folder/mkosi")
+    assert backend.workspace(Path("/home/../home/folder/mkosi/mkosi")) == Path("/home/../home/folder/mkosi")
+    assert backend.workspace(Path("/")) == Path("/")
+    assert backend.workspace(Path()) == Path()
+
```

Comment:
Small style nit: Here and between all top-level functions and classes two empty lines.